recipes-openxt/manager/xenmgr/xenmgr.initscript

```diff
@@ -17,90 +17,84 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 #
 
-# (use /bin/sh when watchdog stuff is removed)
-
-# Provides:		xenmgr
+### BEGIN INIT INFO
+# Provides:             xenmgr
+# Required-Start:       dbus-1 dbd
+# Required-Stop:
+# Default-Start:        5
+# Default-Stop:         0 1 6
+# Short-Description:    Starts OpenXT Xen manager.
+### END INIT INFO
+
+DESC="XenMgr"
+EXEC="/usr/bin/xenmgr"
+PIDFILE="/var/run/xenmgr.pid"
+OPTS=("--writepid=$PIDFILE" "--no-daemonize")
+
+. /etc/init.d/functions-selinux
 
 set -e
 
-test -x /usr/bin/xenmgr || exit 0
-
-export XENVM_DEBUG=syslog:xenvm
-
-XENMGR_OPTS="--no-daemonize"
-if [ -n "$2" ]; then
-    XENMGR_OPTS="$XENMGR_OPTS $2"
-fi
-
-export PATH="${PATH:+$PATH:}/usr/sbin:/sbin"
-
-autostart=0
-
-for i in `cat /proc/cmdline`
-do
-	case "$i" in
-		autostart)
-			autostart=1
-			;;
-		rm_uivm_suspend_img)
-			rm -f /storage/uivm/uivm-suspend-image
-			;;
-	esac
-done
+# Handle options passed by the bootloader on the kernel cmdline.
+cmdline_options() {
+    read -r -a cmdline_opts < /proc/cmdline
+    for i in "${cmdline_opts[@]}"; do
+        case "$i" in
+            no-autostart)
```

Comment:
You flipped the matched string from `autostart` to `no-autostart`, but the grub.cfg/openxt.cfg only add `autostart` or nothing.  Now VMs are always started for regular or console boots.

Defaulting to autostart and disabling on no-autostart makes sense, but that isn't backwards compatible.  I guess that is fine, but "everyone" will have to update any customized openxt.cfg/grub.cfg files.

Reply:
That qualifies as a regression, I can open a new PR and change it back.